File: OrbitalWatch.Api/Models/Satellite.cs
namespace OrbitalWatch.Api.Models;

public class Satellite
{
  public int Id { get; set; }
  public string Name { get; set; } = string.Empty;
  public string NoradId { get; set; } = string.Empty; // e.g. "25544" for ISS
  public string OrbitalRegime { get; set; } = string.Empty; // LEO, MEO, GEO, HEO
  public string Owner { get; set; } = string.Empty;
  public bool IsActive { get; set; } = true;
  public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

  public ICollection<TelemetryEvent> TelemetryEvents { get; set; } = new List<TelemetryEvent>();
  public ICollection<Maneuver> Maneuvers { get; set; } = new List<Maneuver>();
}

File: orbital-watch-web/src/types/Satellite.ts
export interface Satellite {
    id: number
    name: string
    noradId: string
    orbitalRegime: 'LEO' | 'MEO' | 'GEO' | 'HEO'
    owner: string
    isActive: boolean
}

export interface Telemetry {
    id: number
    satelliteId: number
    timestampt: string      // ISO 8601 from .NET DateTime
    altitudeDeg: number
    longitudeDeg: number
    latitudeDeg: number
    velocityXKms: number
    velocityYKms: number
    velocityZKms: number
    speedKms: number
}

export interface ConjunctionAlert {
    primarySatelliteId: number
    secondarySatelliteId: number
    missDistanceKm: number
    detectedAt: string
    severity: 'LOW' | 'MEDIUM' | 'HIGH' | 'CRITICAL'
}

export type ConnectionStatus = 'disconnected' | 'connecting' | 'connected' | 'reconnecting'